quickwit/quickwit-config/src/source_config/mod.rs

```diff
@@ -225,59 +230,113 @@ pub enum SourceParams {
     #[serde(rename = "pubsub")]
     PubSub(PubSubSourceParams),
     Pulsar(PulsarSourceParams),
+    Stdin,
+    Sqs(SqsSourceParams),
     Vec(VecSourceParams),
     Void(VoidSourceParams),
 }
 
 impl SourceParams {
-    pub fn file<P: AsRef<Path>>(filepath: P) -> Self {
-        Self::File(FileSourceParams::file(filepath))
+    pub fn file_from_uri(uri: Uri) -> Self {
+        Self::File(FileSourceParams::Filepath(uri))
+    }
+
+    pub fn file_from_str<P: AsRef<str>>(filepath: P) -> anyhow::Result<Self> {
+        Uri::from_str(filepath.as_ref()).map(Self::file_from_uri)
     }
 
     pub fn stdin() -> Self {
-        Self::File(FileSourceParams::stdin())
+        Self::Stdin
     }
 
     pub fn void() -> Self {
         Self::Void(VoidSourceParams)
     }
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize, utoipa::ToSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum FileSourceMessageType {
+    /// See <https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html>
+    S3Notification,
+    /// A string with the URI of the file (e.g `s3://bucket/key`)
+    RawUri,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize, utoipa::ToSchema)]
+pub struct FileSourceSqs {
+    pub queue_url: String,
+    pub message_type: FileSourceMessageType,
+}
+
 #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize, utoipa::ToSchema)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum FileSourceNotification {
+    Sqs(FileSourceSqs),
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize, utoipa::ToSchema)]
 #[serde(deny_unknown_fields)]
-pub struct FileSourceParams {
-    /// Path of the file to read. Assume stdin if None.
-    #[schema(value_type = String)]
-    #[serde(skip_serializing_if = "Option::is_none")]
-    #[serde(default)]
-    #[serde(deserialize_with = "absolute_filepath_from_str")]
-    pub filepath: Option<PathBuf>, //< If None read from stdin.
+pub struct FileSourceParamsInner {
```

Comment:
I don't think this needs to be pub.
The name is wrong too.

`FileSourceParamsInner` -> `FileSourceParamsForSerialization`

The member also do not need to be pub.

Reply:
I was mimicking `KinesisSourceParamsInner` nearby, but I agree with all these comments!